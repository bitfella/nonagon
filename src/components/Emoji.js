import React from 'react';
import PropTypes from 'prop-types';

const Emoji = ({ label, symbol }) => (
  <span role="img" aria-label={label}>
    {symbol}
  </span>
);

Emoji.propTypes = {
  label: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired
};

export default Emoji;
