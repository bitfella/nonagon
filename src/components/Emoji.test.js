import React from 'react';
import { shallow } from 'enzyme';
import Emoji from './Emoji';

const props = { symbol: 'ðŸ¤˜', label: 'rock on' };
let wrapper;

beforeEach(() => {
  wrapper = shallow(<Emoji {...props} />);
});

describe('Emoji Component', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('renders role attribute with "img" value if correct props are passed in', () => {
    expect(wrapper.find('span').props()['role']).toBe('img');
  });

  it('renders aria label attribute if correct props are passed in', () => {
    expect(wrapper.find('span').props()['aria-label']).toBeDefined();
  });
});
