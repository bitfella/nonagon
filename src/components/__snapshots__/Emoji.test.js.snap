// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji Component renders correctly 1`] = `
<span
  aria-label="rock on"
  role="img"
>
  ðŸ¤˜
</span>
`;
